File: C++/Problems/Problems/problem_3179.cpp
#include "problem_3179.h"

bool problem_3179::test() {
    int n = 4;
    int k = 5;

    int expected = 56;

    auto result = valueAfterKSeconds(n, k);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_3179::valueAfterKSeconds(int n, int k) {
    std::vector<int> arr(n, 1);
    int mod = 1000000007;
    for (int i = 0; i < k; ++i)
        for (int j = 1; j < n; ++j)
            arr[j] = (arr[j - 1] + arr[j]) % mod;
    return arr[n - 1];
}

File: Python/Problems/Problems/problem3179.py
import heapq
import math
import sys

import numpy as np

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem

class Problem3179(Problem):
    def test(self) -> bool:
        n = 4
        k = 5

        expected = 56

        result = self.valueAfterKSeconds(n, k)

        print(result)

        return result == expected

    def valueAfterKSeconds(self, n: int, k: int) -> int:
        arr = [1] * n
        mod = 1000000007
        for i in range(k):
            for j in range(1, n):
                arr[j] = (arr[j - 1] + arr[j]) % mod
        return arr[n - 1]